src/interfaces/particle_distribution.jl

```diff
@@ -0,0 +1,244 @@
+
+# todo: find better name for variate forms used in QEDevents.jl
+abstract type QEDlikeVariate <: Distributions.VariateForm end
+
+"""
+    ParticleLikeVariate{N<:Int}
+
+Auxiliary type to represent the number of axes in the space of all particles:
+
+* `N == 0` single particle vairate
+* `N == 1` multiple particle variate
+
+"""
+abstract type ParticleLikeVariate{N} <: QEDlikeVariate end
+
+"""
+
+    SingleParticleVariate
+
+Auxiliary type to represent single particles in the context of distributions and sampler.
+A sample from this variate form has the type `ParticleStateful`.
+
+"""
+const SingleParticleVariate = ParticleLikeVariate{0}
+
+"""
+
+   MultiParticleVariate
+
+Auxiliary type to represent multiple particles in the context of distributions and sampler.
+A sample from this variate form has the type `Vector{ParticleStateful}`.
+
+"""
+const MultiParticleVariate = ParticleLikeVariate{1}
```

Comment:
Usually, I use type hierarchies, if there are common functions on certain levels. Here, `SingleParticleVariate` and `MultiParticleVariate` do not need a common ground, because they are just *scalar* and *vector* of the same thing, particles.